src/translator/AbstractTranslationModel.cpp

```diff
@@ -6,16 +6,19 @@
 
 // All 3rd party includes
 #include "3rd_party/marian-dev/src/common/options.h"
+#include "3rd_party/marian-dev/src/common/config_parser.h"
+#include "3rd_party/marian-dev/src/3rd_party/yaml-cpp/yaml.h"
 
 // All local includes
 #include "AbstractTranslationModel.h"
 #include "TranslationModel.h"
-#include "TranslationModelConfigToOptionsAdaptor.h"
 
 
 std::shared_ptr<AbstractTranslationModel>
-AbstractTranslationModel::createInstance(const TranslationModelConfiguration& config) {
-	TranslationModelConfigToOptionsAdaptor adaptor;
-	auto options = adaptor.adapt(config);
+AbstractTranslationModel::createInstance(const std::string& config) {
+	auto options = std::make_shared<marian::Options>(YAML::Load(config));
+	marian::ConfigParser configParser(marian::cli::mode::translation);
+	const YAML::Node& defaultConfig = configParser.getConfig();
+	options->merge(defaultConfig);
 	return std::make_shared<TranslationModel>(options);
```

Comment:
Just pass the string down, why do you need `marian::` stuff here? This is supposed to construct other (non-marian) translators which implement AbstractTranslationModel as well, correct? Let the implementation handle how to do configs. Add a translator tag based on which the TranslationModel is built and use the tag to switch. If unknown, raise an unknown translator error.